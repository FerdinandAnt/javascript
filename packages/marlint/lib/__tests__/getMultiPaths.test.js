const getMultiPaths = require('../getMultiPaths');

test('convert array of path into array of array of path', () => {
  const paths = [
    'packages/eslint-config-marlint/src/index.js',
    'packages/eslint-config-marlint/src/rules/base.js',
    'packages/marlint/src/index.js',
    'packages/marlint/src/cli.js',
    'packages/atom.marlint/src/index.js',
    'packages/atom.marlint/src/linter.js',
  ];

  const multiPaths = getMultiPaths(paths);
  expect(multiPaths).toEqual([
    [
      'packages/eslint-config-marlint/src/index.js',
      'packages/eslint-config-marlint/src/rules/base.js',
    ],
    ['packages/marlint/src/index.js', 'packages/marlint/src/cli.js'],
    [
      'packages/atom.marlint/src/index.js',
      'packages/atom.marlint/src/linter.js',
    ],
  ]);
});

test('works on hybrid paths combination', () => {
  const paths = [
    'packages/eslint-config-marlint/src/index.js',
    'packages/eslint-config-marlint/src/rules/base.js',
    'marlint/src/index.js',
    'marlint/src/cli.js',
  ];

  const multiPaths = getMultiPaths(paths);
  expect(multiPaths).toEqual([
    [
      'packages/eslint-config-marlint/src/index.js',
      'packages/eslint-config-marlint/src/rules/base.js',
    ],
    ['marlint/src/index.js', 'marlint/src/cli.js'],
  ]);
});
