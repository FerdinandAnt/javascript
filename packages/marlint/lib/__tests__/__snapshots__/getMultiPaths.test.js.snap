// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert array of path into array of array of path 1`] = `
Array [
  Object {
    "options": undefined,
    "paths": Array [
      "packages/eslint-config-marlint/src/index.js",
      "packages/eslint-config-marlint/src/rules/base.js",
    ],
  },
  Object {
    "options": undefined,
    "paths": Array [
      "packages/marlint/src/index.js",
      "packages/marlint/src/cli.js",
    ],
  },
  Object {
    "options": undefined,
    "paths": Array [
      "packages/atom.marlint/src/index.js",
      "packages/atom.marlint/src/linter.js",
    ],
  },
]
`;

exports[`works on hybrid paths combination 1`] = `
Array [
  Object {
    "options": undefined,
    "paths": Array [
      "packages/eslint-config-marlint/src/index.js",
      "packages/eslint-config-marlint/src/rules/base.js",
    ],
  },
  Object {
    "options": undefined,
    "paths": Array [
      "marlint/src/index.js",
      "marlint/src/cli.js",
    ],
  },
]
`;
